src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -64270,6 +64270,438 @@ public CustomHandler(int literalLength, int formattedCount" + (validityParameter
+                // M(s, $""); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "s").WithArguments("t", "CustomHandler.CustomHandler(int literalLength, int formattedCount, string t)").WithLocation(7, 3)
+            );
+        }
+
+        [Fact, WorkItem(54583, "https://github.com/dotnet/roslyn/issues/54583")]
+        public void StringInterpolation_09()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+C? c = null;
+c.M($"""");
+
+public class C
+{
+    public void M([InterpolatedStringHandlerArgument("""")] CustomHandler c) {}
+}
+
+public partial struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount, C c) : this()
+    {
+    }
+}
+";
+            var handler = GetInterpolatedStringCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: false);
+
+            var comp = CreateCompilation(new[] { code, handler, InterpolatedStringHandlerArgumentAttribute });
+            comp.VerifyDiagnostics(
+                // (7,1): warning CS8602: Dereference of a possibly null reference.
+                // c.M($"");
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c").WithLocation(7, 1)
+            );
+        }
+
+        [Fact, WorkItem(54583, "https://github.com/dotnet/roslyn/issues/54583")]
+        public void StringInterpolation_10()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+C<string?> c = null!;
+c.M($""""); // 1
+c!.M($"""");
+
+public class C<T>
+{
+    public void M([InterpolatedStringHandlerArgument("""")] CustomHandler c) {}
+}
+
+public partial struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount, C<string> c) : this()
+    {
+    }
+}
+";
+            var handler = GetInterpolatedStringCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: false);
+
+            var comp = CreateCompilation(new[] { code, handler, InterpolatedStringHandlerArgumentAttribute });
+            comp.VerifyDiagnostics(
+                // (7,1): warning CS8620: Argument of type 'C<string?>' cannot be used for parameter 'c' of type 'C<string>' in 'CustomHandler.CustomHandler(int literalLength, int formattedCount, C<string> c)' due to differences in the nullability of reference types.
+                // c.M($""); // 1
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "c").WithArguments("C<string?>", "C<string>", "c", "CustomHandler.CustomHandler(int literalLength, int formattedCount, C<string> c)").WithLocation(7, 1)
+            );
+        }
+
+        [Fact, WorkItem(54583, "https://github.com/dotnet/roslyn/issues/54583")]
+        public void StringInterpolation_11()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+M(out string? s1, $"""");
+
+void M(out string? t1, [InterpolatedStringHandlerArgument(""t1"")] CustomHandler c) => t1 = null;
```

Comment:
I feel silly for asking at this stage but.. I am wondering what is supposed to happen when the user sees this in intellisense? Are most APIs going to be overloaded on 'string' and 'CustomHandler' in practice so the user will see the string overload and say "ok, I pretty much know what to provide here"?

Reply:
Likely most will overload, but nothing special will be shown for custom handlers.